synapse/handlers/room_list.py

```diff
@@ -73,8 +74,14 @@ def get_local_public_room_list(self, limit=None, since_token=None,
             # We explicitly don't bother caching searches or requests for
             # appservice specific lists.
             logger.info("Bypassing cache as search request.")
+
+            # XXX: Quick hack to stop room directory queries taking too long.
+            # Timeout request after 60s. Probably want a more fundamental
+            # solution at some point
+            timeout = datetime.now() + timedelta(seconds=60)
```

Comment:
For the record, I'm not generally a fan of datetime objects when a simple "seconds since the epoch" will do and be much more explicit. They have a whole bunch of arithmetic and timezone logic which is pretty redundant here. (In this case, `self.clock.time() + 60` will do what you want).

I'm happy if you want to leave this though, given it's a quick hack.

Reply:
(as a question to ponder: what will this do when the clocks change? I *think* it will do the right thing, but tbh who knows. It's a question you just don't need to worry about with "seconds since the epoch".)

Reply:
I've changed it to just use epoch seconds, thanks for the heads up :)